python/ray/train/v2/_internal/callbacks/state_manager.py

```diff
@@ -152,3 +150,13 @@ def before_worker_group_shutdown(self, worker_group: WorkerGroup):
                 run_id=self._run_id,
                 attempt_id=worker_group_context.run_attempt_id,
             )
+
+    def before_controller_abort(
+        self, worker_group_context: Optional[WorkerGroupContext]
+    ):
+        self._state_manager.update_train_run_aborted(self._run_id)
+        if worker_group_context:
+            self._state_manager.update_train_run_attempt_aborted(
+                self._run_id,
+                worker_group_context.run_attempt_id,
+            )
```

Comment:
Hmm, it's a little strange that this hook for the controller is providing the worker group information. Ideally those are self-contained in the `WorkerGroupCallback`s.

Here's a proposal:

```python
class Controller:
    async def abort(self):
        # This triggers the after_controller_state_update,
        # and we can actually just include the `update_train_run_aborted` call in the elif chain above.
        self._set_state(ABORTED).
        if self._worker_group:
            # This can trigger some WorkerGroupCallback to update the train run attempt status to Aborted
            self._worker_group.abort()
```

Reply:
Technically the `before_controller_abort` would not be needed in this PR anymore -- we can remove and add it in later as needed.

Reply:
The self-containment looks very clean to me!

Reply:
Sure, I think this makes sense. Just to clarify, this means that `ControllerCallback` won't need `before_controller_abort` because `after_controller_state_update` covers it, but `WorkerGroupCallback` does need `before_worker_group_abort`, right?

Reply:
yeah that's right. The worker group doesn't have its own state abstraction like the controller does right now